projects/datetime-picker/src/lib/datepicker-input.ts

```diff
@@ -74,8 +74,10 @@ export class NgxMatDatepickerInput<D>
   }
   set min(value: D | null) {
     const validValue = this._dateAdapter.getValidDateOrNull(this._dateAdapter.deserialize(value));
+    const valueTimestamp = new Date(validValue as Date | string).getTime();
+    const minTimestamp = new Date(this._min as Date | string).getTime();
 
-    if (!this._dateAdapter.sameDate(validValue, this._min)) {
+    if (valueTimestamp !== minTimestamp){
```

Comment:
Not using the date adapter anymore I think is a mistake and a violation of the concept of the use of the date adapter.